app/controllers/users_controller.rb

```diff
@@ -0,0 +1,47 @@
+class UsersController < ApplicationController
+  def index
+    @user = User.all
+  end
+
+  def show
+    @user = User.find_by(id: params[:id])
+  end
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    # @user = User.new(username: params[:username], email: params[:email], password: params[:password])
+
+    if @user.save
+      redirect_to user_path(@user.id)
+    else
+      render :new
+    end
+  end
+
+  def update
+    @user = User.find(params[:id])
+    @user.update(user_params)
+    @user.save
+    redirect_to user_path(@user.id)
+  end
```

Comment:
Please make sure to `redirect_to` **only** if `@user.save` was successful. Otherwise, just `render :edit`. You may use an `if` statement for this.